app/javascript/listings/components/ListingTagsField.jsx

```diff
@@ -0,0 +1,121 @@
+import { h } from 'preact';
+import { useEffect, useMemo, useState } from 'preact/hooks';
+import PropTypes from 'prop-types';
+import { useTagsField } from '../../hooks/useTagsField';
+import { TagAutocompleteOption } from '../../article-form/components/TagAutocompleteOption';
+import { TagAutocompleteSelection } from '../../article-form/components/TagAutocompleteSelection';
+import { MultiSelectAutocomplete } from '@crayons';
+
+export const ListingTagsField = ({
+  defaultValue,
+  onInput,
+  categorySlug,
+  name,
+  onFocus,
+}) => {
+  const listingState = useMemo(
+    () => ({
+      additionalTags: {
+        jobs: [
+          'remote',
+          'remoteoptional',
+          'lgbtbenefits',
+          'greencard',
+          'senior',
+          'junior',
+          'intermediate',
+          '401k',
+          'fulltime',
+          'contract',
+          'temp',
+        ],
+        forhire: [
+          'remote',
+          'remoteoptional',
+          'lgbtbenefits',
+          'greencard',
+          'senior',
+          'junior',
+          'intermediate',
+          '401k',
+          'fulltime',
+          'contract',
+          'temp',
+        ],
+        forsale: ['laptop', 'desktopcomputer', 'new', 'used'],
+        events: ['conference', 'meetup'],
+        collabs: ['paid', 'temp'],
+      },
+    }),
+    [],
+  );
+
+  const { defaultSelections, topTags, fetchSuggestions, syncSelections } =
+    useTagsField({ defaultValue, onInput });
+  const [topTagsWithAdditional, setTopTagsWithAdditional] = useState([
+    ...topTags,
+  ]);
+
+  useEffect(() => {
+    if (listingState && listingState.additionalTags) {
+      const newTagsForSelectedCategory =
+        listingState.additionalTags[categorySlug] || [];
+      // Push in this way: { name: 'remote' }
+      const formattedNewTagsForSelectedCategory =
+        newTagsForSelectedCategory.map((name) => ({ name }));
+      setTopTagsWithAdditional([
+        ...topTags,
+        ...formattedNewTagsForSelectedCategory,
+      ]);
+    }
+  }, [topTags, listingState, categorySlug]);
+
+  const fetchSuggestionsWithAdditionalTags = async (searchTerm) => {
+    const suggestionsResult = await fetchSuggestions(searchTerm);
+    const suggestedNames = suggestionsResult.map((t) => t.name);
+
+    // Search in the topTagsWithAdditional array
+    const additionalItems = topTagsWithAdditional.filter((t) =>
+      t.name.startsWith(searchTerm),
+    );
+    // Remove duplicates
+    additionalItems.forEach((t) => {
+      if (!suggestedNames.includes(t.name)) {
+        suggestionsResult.push(t);
+      }
+    });
+    return suggestionsResult;
+  };
```

Comment:
(refactor / blocking) - I think we need to make sure the results are ordered by name after we insert these additional items. At the moment the category-specific ones are appended to the end, and it looks a bit funky that they're not all in alphabetical order e.g:
![tag suggestions not in alphabetical order](https://user-images.githubusercontent.com/20773163/158787329-acef0ae2-8136-479b-a982-5aeab37d6dcd.png)

Reply:
Done!